astropy/cosmology/_utils.py

```diff
@@ -71,3 +75,9 @@ def aszarr(z):
         return z
     # not one of the preferred types: Number / array ducktype
     return Quantity(z, cu.redshift).value
+
+
+def all_cls_vars(cls: object | type) -> dict[str, Any]:
+    """Return all variables in the whole class hierarchy."""
+    kls = cls if isinstance(cls, type) else type(cls)
+    return functools.reduce(operator.__or__, map(vars, kls.mro()[::-1]))
```

Comment:
If you want to have atomic commits then this function should not be edited by two commits.

Reply:
Done!